Makefile

```
@@ -46,7 +46,10 @@ LDFLAGS := -X github.com//open-telemetry/opentelemetry-ebpf-profiler/vc.version=
 
 GO_FLAGS := -buildvcs=false -ldflags="$(LDFLAGS)" -tags osusergo,netgo
 
-all: generate ebpf binary
+MAKEFLAGS += -j$(shell nproc)
+
+all: generate ebpf
```

Why is `binary` removed? Looks to me like the actual agent is no longer being built? Or it's now built somewhere else?

---

I believe because `binary` alone does not do much?
It is `binaries` that builds all the binaries AFAIR

---

So maybe we want this?

```suggestion
all: generate ebpf binaries
```

---

This is incorrect (as was the previous `all` target with a parallel Make) as there is no fixed order of dependencies. This is why (I'm assuming) @rockdaboot removed the `binary` target altogether. The agent binary will now be built through `all`. The caveat is that we don't have a target to only build the Go binary without building its dependencies (`generate`, `ebpf`) every time.

On my system this takes 2 seconds so it's not a big deal, but if we can always introduce an additional target just for the Go binary. Ideally, at some point we'll move to a proper Make build where all targets (ebpf, Go) and their dependencies are declaratively subsumed/tracked by `Make`.

---

No, I'd keep the `binary` target still. The `binary` target should depend on `ebpf` since it now uses the ebpf files via embed

---

Or rename `all` to something else. Perhaps that's just a nitpick on wording then.
Also, just now realized the `go build` command was moved from `binary` to `all`. I didn't see that earlier.

---

@fabled and I agreed on the following in a private chat:
```
all: ebpf-profiler

ebpf-profiler: generate ebpf
```

This is extendable, in case we have more binaries to be built with `make`.
Also, it matches the binary file name which is more intuitive/clear in comparison with `binary`.